docs: Audit and streamline CONTRIBUTING.md (#5595)
typescript-eslint/typescript-eslint: CONTRIBUTING.md

Feel free to raise an issue if you have a question, an enhancement, or a bug report.

Use the issue search functionality to search all **_opened and closed_** [issues](https://github.com/typescript-eslint/typescript-eslint/issues) before raising a new issue. If you raise a duplicate issue, you're just creating noise for everyone watching this repo.

Before raising a bug, ensure you are using the latest version of our packages. We release every week, so there's a good chance your issue might have already been fixed.

Finally, when raising a new issue, please fill out the [issue template](https://github.com/typescript-eslint/typescript-eslint/issues/new/choose) - **_please don't skip sections_**.

Please provide **_as much information as possible_**. This project is maintained by volunteers, so the more information you provide, the less likely we will have to waste everyone's time in asking you for more information.

> If you have a particularly complex issue - consider creating a small, self-contained reproduction repo or use our [playground](https://typescript-eslint.io/play). This will help you in figuring out the exact problem, and will help us in reproducing and diagnosing the bug.

Reviewer: The complex issue template is meant for things that can't be demonstrated on the playground. +1 to mentioning it though. How does this strike you?

```suggestion
If you have a particularly complex issue that can't be reproduced on [our playground](https://typescript-eslint.io/play) - consider creating a small, self-contained reproduction repo or use. This will help you in figuring out the exact problem, and will help us in reproducing and diagnosing the bug.
```

**_Help us to help you_**

## Questions and requests for support

Questions and requests for support should not be opened as issues and should be handled in the following ways:

- Ask a question on [StackOverflow](https://stackoverflow.com/questions/tagged/typescript-eslint) using the `typescript-eslint` tag.

> - Join our [Discord server](https://discord.com/invite/??).

Reviewer: Indeed, we don't have one. We're not paid enough to deal with a flood of realtime questions from the public. 😄

Pending what the other maintainers think, I'm up for directing people to ask on Twitter instead. I'm pretty active there.

```suggestion
- Publicly tweet [@tseslint on Twitter](https://twitter.com/tseslint).
```

Once your changes are ready, you can raise a PR. The title of your PR should match the following format:

```text

> <type>(<package>): <Subject> (#Issue Number)

Reviewer: The `(#Issue Number)` gets added by GitHub. We don't need it.

Also, I kind of like _short description_ more than `Subject` TBH. It's more clear what it's supposed to be: short, not a whole long ordeal. Thoughts?

```

> Samples: (even more [samples](https://github.com/typescript-eslint/typescript-eslint/commits/main))

Reviewer: Linking to the latest commits is a good idea. But this sentence is a little oddly worded. Maybe...

```suggestion
You can find more samples of good past PR titles in [recent commits to `main`](https://github.com/typescript-eslint/typescript-eslint/commits/main))
```

Related: it's an accessibility necessity to have informative link titles. When screenreader folks navigate through the page by skipping through links, they don't receive context around the links. Titles like _"samples"_ don't make much sense out of context.

```text
fix(scope-manager): correct handling for class static blocks
```

```text
docs: Fix links to getting started in README.md
```

> ### Revert
> 
> If the commit reverts a previous commit, it should begin with `revert:`, followed by the header of the reverted commit. In the body it should say: `This reverts commit <hash>.`, where the hash is the SHA of the commit being reverted.
> 

Reviewer: Contributors don't need to know this. It's a good callout for #5238 though!

```suggestion
```

Please note that as this project is maintained by volunteers, it may take a while for us to get around to your PR (sometimes a month or more). Be patient, we'll get to it. Please refrain from commenting asking for a review, or similar bump comments. **_These just create spam for maintainers, and does not push your PR higher in the queue_**.

Once we have reviewed your PR, we will provide any feedback that needs addressing. If you feel a requested change is wrong, don't be afraid to discuss with us in the comments. Once the feedback is addressed, and the PR is reviewed, we'll ensure the branch is up to date with `main`, and merge it for you.

> 
> ## Financial contributions
> 
> We also welcome financial contributions in full transparency on our [open collective](https://opencollective.com/typescript-eslint).
> Anyone can file an expense. If the expense makes sense for the development of the community, it will be "merged" in the ledger of our open collective by the core contributors and the person who filed the expense will be reimbursed.
> 
> ### Contributors
> 
> Thank you to all the people who have already contributed to typescript-eslint!
> <a href="https://github.com/typescript-eslint/typescript-eslint/graphs/contributors"><img src="https://opencollective.com/typescript-eslint/contributors.svg?width=890" /></a>
> 
> ### Backers
> 
> Thank you to all our backers! [[Become a backer](https://opencollective.com/typescript-eslint#backer)]
> 
> <a href="https://opencollective.com/typescript-eslint#backers" target="_blank"><img src="https://opencollective.com/typescript-eslint/backers.svg?width=890"></a>
> 
> ### Sponsors
> 
> Thank you to all our sponsors! (please ask your company to also support this open source project by [becoming a sponsor](https://opencollective.com/typescript-eslint#sponsor))
> 
> <a href="https://opencollective.com/typescript-eslint/sponsor/0/website" target="_blank"><img src="https://opencollective.com/typescript-eslint/sponsor/0/avatar.svg"></a>
> <a href="https://opencollective.com/typescript-eslint/sponsor/1/website" target="_blank"><img src="https://opencollective.com/typescript-eslint/sponsor/1/avatar.svg"></a>
> <a href="https://opencollective.com/typescript-eslint/sponsor/2/website" target="_blank"><img src="https://opencollective.com/typescript-eslint/sponsor/2/avatar.svg"></a>
> <a href="https://opencollective.com/typescript-eslint/sponsor/3/website" target="_blank"><img src="https://opencollective.com/typescript-eslint/sponsor/3/avatar.svg"></a>
> <a href="https://opencollective.com/typescript-eslint/sponsor/4/website" target="_blank"><img src="https://opencollective.com/typescript-eslint/sponsor/4/avatar.svg"></a>
> <a href="https://opencollective.com/typescript-eslint/sponsor/5/website" target="_blank"><img src="https://opencollective.com/typescript-eslint/sponsor/5/avatar.svg"></a>
> <a href="https://opencollective.com/typescript-eslint/sponsor/6/website" target="_blank"><img src="https://opencollective.com/typescript-eslint/sponsor/6/avatar.svg"></a>
> <a href="https://opencollective.com/typescript-eslint/sponsor/7/website" target="_blank"><img src="https://opencollective.com/typescript-eslint/sponsor/7/avatar.svg"></a>
> <a href="https://opencollective.com/typescript-eslint/sponsor/8/website" target="_blank"><img src="https://opencollective.com/typescript-eslint/sponsor/8/avatar.svg"></a>
> <a href="https://opencollective.com/typescript-eslint/sponsor/9/website" target="_blank"><img src="https://opencollective.com/typescript-eslint/sponsor/9/avatar.svg"></a>

Reviewer: I don't think contributors & sponsors information belongs in our `CONTRIBUTING.md`. We already showcase contributors on the root `README.md` and sponsors on the website. Let's keep each page as small as possible.

```suggestion
```

Author: cool
